acceptance-tests/src/test/scala/ch/datascience/graph/acceptancetests/CommitSyncFlowsSpec.scala

```diff
@@ -0,0 +1,123 @@
+      val missedCommitId    = commitIds.generateOne
+      val committer         = persons.generateOne
+
+      Given("commit with the commit id matching Push Event's 'after' exists on the project in GitLab")
+      `GET <gitlabApi>/projects/:id/repository/commits/:sha returning OK with some event`(projectId, nonMissedCommitId)
+
+      And("fetch latest commit endpoint returns the non missed and later the missed commit")
+      `GET <gitlabApi>/projects/:id/repository/commits returning OK with a commit`(projectId,
+                                                                                   nonMissedCommitId,
+                                                                                   missedCommitId
+      )
+      `GET <gitlabApi>/projects/:id/repository/commits/:sha returning OK with some event`(projectId, missedCommitId)
+
+      And("RDF triples are generated by the Remote Triples Generator for both commits")
+      `GET <triples-generator>/projects/:id/commits/:id returning OK with some triples`(project,
+                                                                                        nonMissedCommitId,
+                                                                                        committer
+      )
+      `GET <triples-generator>/projects/:id/commits/:id returning OK with some triples`(project,
+                                                                                        missedCommitId,
+                                                                                        committer
+      )
+
+      And("access token is present")
+      givenAccessTokenPresentFor(project)
+
+      And("project members/users exists in GitLab")
+      `GET <gitlabApi>/projects/:path/members returning OK with the list of members`(
+        project.path,
+        committer.asMembersList()
+      )
+
+      And("project exists in GitLab")
+      `GET <gitlabApi>/projects/:path returning OK with`(project)
+
+      When("a Push Event arrives for the non missed event")
+      webhookServiceClient
+        .POST("webhooks/events", HookToken(projectId), data.GitLab.pushEvent(project, nonMissedCommitId))
+        .status shouldBe Accepted
+
+      And("relevant commit events are processed")
+      `wait for events to be processed`(project.id)
+
+      Then("triples for the non missed events should be in the RDF Store")
+      RDFStore.commitTriplesCount(nonMissedCommitId) should be > 0
+
+      When("commit synchronisation process kicks-off")
+      db.EventLog.execute {
+        sql"""|DELETE FROM subscription_category_sync_time 
+              |WHERE project_id = ${projectId.value} AND category_name = 'COMMIT_SYNC'
+              |""".stripMargin.update.run
+      }
+      sleep((4 seconds).toMillis)
+
+      And("commit events for the missed event are created and processed")
+      `wait for events to be processed`(project.id)
+
+      Then("triples for the missed commit should be in the RDF Store")
```

Comment:
Perhaps we should also check that if another Push Event arrives, that nothing happens